src/Goteo/Controller/InvestController.php

```diff
@@ -96,11 +97,23 @@ private function validate(
             Config::get('currency')
         );
 
+        if ($request) {
+            $return_to = '';
+
+            if ($request->query->has('return_to')) {
+                $return_to = rawurldecode($request->query->get('return_to'));
```

Comment:
Why not place the url decoding inside the `isAllowedDomain` static method? This way we can safely use the functionality without having to worry about the string we are passing to it.

Reply:
Good catch, thanks!